bns-test/test/namespace_update_price_function.test.ts

```diff
@@ -0,0 +1,181 @@
+          amount: 10_000_000_000_000
+        },
+      ]
+      const binFile = getDefaultBinaryFilePath();
+      const dbFileName = getTempFilePath();
+      provider = await NativeClarityBinProvider.create(allocations, dbFileName, binFile);
+      bns = new BNSClient(provider);
+      await bns.deployContract();
+    });
+  
+    it("Given a revealed pre-order from Alice for the namespace 'blockstack' initiated at block #20", async () => {
+      var receipt = await bns.namespacePreorder(cases[0].namespace, cases[0].salt, cases[0].value, {
+        sender: cases[0].namespaceOwner
+      });
+      expect(receipt.success).eq(true);
+      expect(receipt.result).include('u146');
+  
+      receipt = await bns.namespaceReveal(
+        cases[0].namespace,
+        cases[0].salt,
+        cases[0].priceFunction,
+        cases[0].renewalRule,
+        cases[0].nameImporter, {
+          sender: cases[0].namespaceOwner
+        });
+      expect(receipt.success).eq(true);
+      expect(receipt.result).include('true');
+  
+      // Launching the namespace
+      // should succeed if the namespace has not already been launched, and revealed less than a year ago (todo: fix TTL)
+      receipt = await bns.namespaceReady(cases[0].namespace, {
+        sender: cases[0].namespaceOwner
+      });
+      expect(receipt.success).eq(true);
+      expect(receipt.result).include('true');
+  
+      receipt = await bns.getNamePrice(cases[0].namespace, "a");
+      expect(receipt.success).eq(true);
+      // curl https://core.blockstack.org/v2/prices/names/a.blockstack
+      expect(receipt.result).include(`(ok u20)`); 
+
+      let priceFunctionUpdate1 = {
+        buckets: [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1],
+        base: 1,
+        coeff: 4, // <- update
+        noVowelDiscount: 0,
+        nonAlphaDiscount: 0,
+      };
+
+      // Updating the price function should succeed
+      receipt = await bns.namespaceUpdatePriceFunction(cases[0].namespace, 
+        priceFunctionUpdate1,
+        {
+        sender: cases[0].namespaceOwner
+      });
+      expect(receipt.success).eq(true);
+      expect(receipt.result).include('true');
+
+      // New price should be 2x what we had before
+      receipt = await bns.getNamePrice(cases[0].namespace, "a");
+      expect(receipt.success).eq(true);
+      expect(receipt.result).include(`(ok u40)`); 
+
+        // Now, let's give up on our ability to update the price function
+      receipt = await bns.namespaceRevokePriceFunctionUpdates(cases[0].namespace, 
+        {
+        sender: cases[0].namespaceOwner
+      });
+      expect(receipt.success).eq(true);
+      expect(receipt.result).include('true');
+
+      let priceFunctionUpdate2 = {
+        buckets: [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1],
+        base: 1,
+        coeff: 8, // <- update
+        noVowelDiscount: 0,
+        nonAlphaDiscount: 0,
+      };
+
+      // Updating the price function should succeed
```

Comment:
Shouldn't this be `should fail`?

Reply:
Addressed with https://github.com/blockstack/stacks-blockchain/pull/2286/commits/d9831b806de4414eb16a99c1af714193f6ea9704